### ISSUE TITLE
Coverflow can't install in 21.11.00

### ISSUE THREAD

**Issue by MarbleKuok** [bywatersolutions/koha-plugin-coverflow#50]
After i update my koho to version 21.11.00. I would like to install the coverflow plugin. After uploading KPZ packages,there is nothing happening.But the other plugin is work ,like "Galadriel OPAC Theme plugin" and"Example Kitchen-Sink Plugin" . So how could i fix it .Thanks for answering.

**Comment by MarbleKuok** [comment]
Status update: I can only install the Coverflow version 2.4.24.And the other version still not working. After 2.4.24 installed, there is an error show at the console: `"Failed to load resource: the /api/v1/contrib/coverflow/reports/27 server responded with a status of 500 (Internal Server Error)".` 
And click the link will show `{"error":"Unhandled exception (Undefined subroutine &Koha::Plugin::Com::ByWaterSolutions::CoverFlow::execute_query called at \/var\/lib\/koha\/library\/plugins\/Koha\/Plugin\/Com\/ByWaterSolutions\/CoverFlow.pm line 283.\n)"}`
I think koha has updated the reports formats. So anyone can tell me how to update the coverflow. thanks

**Comment by kidclamp** [comment]
I added a patch, can you try the new release?

**Comment by MarbleKuok** [comment]
@kidclamp Thanks for your great work! I have tried your new release. The new release still not working. After uploading KPZ packages,there is nothing happening.But I can see the file in the plugins dir . And can't see any coverflow api at  /api/v1/.html. 
Any other suggestion? How could i found the log in order to offer more information to debug.Thanks again.

**Comment by kidclamp** [comment]
After install, did you restart plack and memcached to ensure the new routes are installed?

Can you share the configuration you have defined in the plugin?

I am able to install and run the coverflow in 21.11

**Comment by MarbleKuok** [comment]
@kidclamp Yes ,I have restarted the koha use this command`sudo systemctl restart memcached koha-common apache2`

And there is my configuration of my koha setting. 
```
<pluginsdir>/var/lib/koha/library/plugins</pluginsdir> <!-- This entry can be repeated to use multiple directories -->
 <enable_plugins>1</enable_plugins>
```
Any other information you needed? Thank you again for everything you’ve done

**Comment by MarbleKuok** [comment]
And i can see the model at the about page.
![圖片](https://user-images.githubusercontent.com/4158304/146117997-449dbb87-6cc5-4d3f-9f9c-15fb0ce21ab8.png)

**Comment by kidclamp** [comment]
If you visit:
your.library.com/api/v1/.html (replace your.library.com with correct link)
Do you see entries for contrib/coverflow?

Can you paste the 'mapping' from the coverflow configure page?
http://localhost:8081/cgi-bin/koha/plugins/run.pl?class=Koha%3A%3APlugin%3A%3ACom%3A%3AByWaterSolutions%3A%3ACoverFlow&method=configure

And can you paste the html of the coverflow you added? did you add to OpacMainUserBlock in additional contents?

**Comment by gveranis** [comment]
I had the same problem, and on logs had the error "Calling 'upgrade' died for plugin Koha::Plugin::Com::ByWaterSolutions::CoverFlow" , I comment some lines at the end of that function and the installation succeed and after that I comment out the same lines and everything is working for the plugin.

**Comment by MarbleKuok** [comment]
@kidclamp i can't see the contrib/coverflow at api page.

replace my localhost.It showed 'error 500' from the configure page 
http://localhost:8081/cgi-bin/koha/plugins/run.pl?class=Koha%3A%3APlugin%3A%3ACom%3A%3AByWaterSolutions%3A%3ACoverFlow&method=configure

I have added this html ino my OpacMainUserBlock.
`<span id="coverflow">Loading...</span>`

But if i remove the file(Coverflow file and Coverflow.pm) in my host .
![圖片](https://user-images.githubusercontent.com/4158304/146297322-ade6d8f2-d0e3-47f9-9722-7c2ae5d284f2.png)
It will show error.
The error will disappear after i upload the KPZ packages .But i still can't see the coverflow configure page

**Comment by MarbleKuok** [comment]
@gveranis Hi, Thanks for your help.Where can i find the logs. Which lines you have comment? I can't see the coverflow plugin in my plugin page.How can i uninstall the plugin and intall it again? Thanks again

**Comment by mosheys** [comment]
> 
Hi I had the same issue and based on @gveranis comment I played around with 
/var/lib/koha/library/plugins/Koha/Plugin/Com/ByWaterSolutions/CoverFlow.pm I commented out all lines
from line # 207 'sub upgrade' to line # 231 'return 1;' included line #  231, then i restarted plack,  koha-plack --restart (your_Instance)

**Comment by MarbleKuok** [comment]
@mosheys it works! Thanks for your help .

**Comment by AndrewIsh** [comment]
I'm going to re-open this as it's is definitely still a problem, though I've not been able to determine why as it was failing on a live system and I couldn't keep restarting Plack to get to the bottom of it.

In the end I had to download the latest package (current 2.5.0), comment out the contents of the `upgrade` function in `CoverFlow.pm`, then the upgrade would proceed.

**Comment by AndrewIsh** [comment]
Ah, seems commenting has not re-opened it, hopefully some kindly soul will do so!

**Comment by kidclamp** [comment]
Possibly resolved by 0d4c92c7602e0922dc5e2aeba06e423c9a50bede?
